src/utils/classes/patternFactory.ts

```diff
@@ -0,0 +1,56 @@
+import { IPatternSettings, MetronomeSounds, SubDivision } from "./models-interfaces";
+import { Pattern } from "./pattern";
+import { getBlankString, getIndexForFillCharacter, getNth, replaceCharacter, replaceEachNthChar } from "./patternUtils";
+
+export class PatternFactory extends Pattern {
+
+	private subDivision: SubDivision;
+
+	constructor(patternSettings?: IPatternSettings, subDivision: SubDivision = 8) {
```

Comment:
can you please add the singleton logic back into the constructor if possible? this class should only ever be instantiated once.

Reply:
@dcallus can make it singleton.

Reply:
will do